pkg/database/user.go

```diff
@@ -156,6 +179,115 @@ func (db *Database) PurgeUsers(maxAge time.Duration) (int64, error) {
 		maxAge = -1 * maxAge
 	}
 	deleteBefore := time.Now().UTC().Add(maxAge)
-	rtn := db.db.Unscoped().Where("disabled = ? and updated_at < ?", true, deleteBefore).Delete(&User{})
+	var recvUsers []*User
+	scope := db.db.Unscoped()
+	whereClause := "disabled = ? and updated_at < ?"
+	if err := scope.Where(whereClause, true, deleteBefore).Find(&recvUsers).Error; err == nil {
+		// Also clear our Redis caches of the respective records.
+		defer db.purgeUsersFromCache(context.Background(), recvUsers...)
+	}
+	rtn := scope.Where(whereClause, true, deleteBefore).Delete(&User{})
 	return rtn.RowsAffected, rtn.Error
 }
+
+// This TTL is set to 1 day so that in the worst case, if a deletion happens but
+// doesn't clear the cache, it'll eventually expire, which ensures eventual "deletion".
+const _CACHE_PURGE_TTL_1DAY = 24 * time.Hour
+
+func (db *Database) cacheUser(ctx context.Context, user *User) error {
+	conn, err := db.getRedisConn(ctx)
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	jsonBlob, err := json.Marshal(user)
+	if err != nil {
+		return err
+	}
+
+	conn.SendContext(ctx, "MULTI")
+	conn.SendContext(ctx, "SET", _USERS_TABLE_BY_EMAIL, user.Email, jsonBlob)
+	conn.SendContext(ctx, "EXPIRE", _USERS_TABLE_BY_EMAIL, user.Email, _CACHE_PURGE_TTL_1DAY.Seconds())
+	conn.SendContext(ctx, "SET", _USERS_TABLE_BY_ID, user.ID, jsonBlob)
+	conn.SendContext(ctx, "EXPIRE", _USERS_TABLE_BY_ID, user.ID, _CACHE_PURGE_TTL_1DAY.Seconds())
+	// Use a Time to live (TTL) of 1days so that for example if succeeded but clearing it from the
```

Comment:
NIT: `Use a Time to live (TTL) of 24 hours so that, for example, if <save?> succeeded but clearing it from the Redis cache failed, the cached information will at least be purged if not used within within 24 hours.`